source/views/building-hours/report/overview.tsx

```diff
@@ -332,4 +336,6 @@ export const NavigationKey = 'BuildingHoursProblemReport'
 
 export const NavigationOptions: NativeStackNavigationOptions = {
 	title: 'Report a Problem',
+	presentation: 'modal',
+	headerRight: () => <CloseScreenButton title="Close" />,
```

Comment:
I'd like to take a vote on whether the overview (second screenshot) should be a modal. We would need more precautions to prevent accidental swipe-to-dismiss if there's changed data in the form? Thoughts?

Reply:
I think it should be a modal. Thoughts on labelling it "Discard" instead of "Close"?

Reply:
Discard seems good to me. Do we want to prevent closing without prompting if changes have been made to the form before discarding?

Reply:
I think that would be best

Reply:
My initial attempt at this is to compare two json objects to each other. This works in some cases but not in all.

It fails to work for areas where we've added to the edited data in the following way:

Take either notes or closed-for-chapel. If we edit a note, the field and value get added but never removed to the object.

Same with the hours toggle -- we toggle on/off and the field remains false in the data, not removed.

Reply:
Do we need a "default" object that we can merge the original / edited objects into, so that all properties are guaranteed to exist?

What if we dump both to the yaml strings, like we're sending them to the server? That strips the optional properties, right?

Reply:
Yeah I was thinking the same thing along the lines of a default object where optional props do not change the shape of the diff.

I can give your idea of dumping the yaml a try.

Reply:
Nice! That worked great and was added to [`4a2c39`](https://github.com/StoDevX/AAO-React-Native/pull/6154/commits/4a2c39a286f3021c1fb8708bbbe4a8eeed559144).